<launch>

  <!-- Default values -->
  <arg name="nodelet_manager_name" default="standalone_nodelet"/>
  <arg name="start_nodelet_manager" default="true"/>
  <arg name="debug_mode" default="false"/>

  <!-- Config Files -->
  <arg name="map_config" default="$(find stargazer_ros_tool)/param/map.yaml"/>
  <arg name="detection_config" default="$(find stargazer_ros_tool)/param/landmark_finder.yaml"/>

  <!-- Topics -->
  <arg name="undistorted_image_topic" default="/camera_top/image_raw"/>
  <arg name="landmark_topic" default="/landmarks_seen"/>

  <!-- Nodelet manager (if enabled) -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"
        if="$(arg start_nodelet_manager)"/>

  <node pkg="nodelet" type="nodelet" name="landmark_finder_nodelet"
        args="load stargazer_ros_tool/LandmarkFinderInterfaceNodelet $(arg nodelet_manager_name)" output="screen">
    <param name="map_config" value="$(arg map_config)"/>
    <param name="landmark_topic" value="$(arg landmark_topic)"/>
    <param name="undistorted_image_topic" value="$(arg undistorted_image_topic)"/>
    <param name="debug_mode" value="$(arg debug_mode)"/>
    <rosparam file="$(arg detection_config)" command="load"/>
  </node>
</launch>
